### ISSUE TITLE
Bug: build fail on ubuntu jammy with utest compilation

### ISSUE THREAD

**Issue by RedLeader962** [norlab-ulaval/libpointmatcher#552]
#### REF:
- NorLab TeamCity build server (require vpn access):  [Build #14](http://132.203.26.125:8111/buildConfiguration/LibpointmatcherBuildSystem_BuildMatrixStatusBuildAndTestMultiArchCiBuildAndTestDockerImages/703?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildProblemsSection=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=false&logView=flowAware) 
- Build log: [Libpointmatcher_Bleeding_build_matrix_build_and_test_multi-arch_ci_PR_docker_images_14 1.log](https://github.com/norlab-ulaval/libpointmatcher/files/13979654/Libpointmatcher_Bleeding_build_matrix_build_and_test_multi-arch_ci_PR_docker_images_14.1.log)
- NorLab YouTrack task [NMO-305](https://norlab.youtrack.cloud/issue/NMO-305) fix: build fail › ubuntu jammy with utest compilation

<br>

Notes:  
- Might be address by solving issue #534 

## Setup:

This command require multi-architecture docker builder set with linux/amd64 and linux arm64 platform but note that the tests are failing on both of them separately as you can see on the build log summary above

```shell
bash build_system/lpm_crawl_libpointmatcher_build_matrix.bleeding.bash \
                                     --repository-version-build-matrix-override latest \
                                     --cmake-build-type-build-matrix-override Release \
                                     --ubuntu-version-build-matrix-override jammy \
                                     --fail-fast -- build ci_PR_arm64v8 ci_PR_amd64
```

```markup
[LPM] Environment variables set for compose:
10:44:14     
10:44:14         LIBPOINTMATCHER_VERSION=latest          
10:44:14         DEPENDENCIES_BASE_IMAGE=ubuntu          
10:44:14         DEPENDENCIES_BASE_IMAGE_TAG=jammy  
10:44:14     
``` 

## Build log summary from NorLab TeamCity build server

![Build_14_Bleeding_build_matrix_build_and_test–multi-arch–ci_PR–docker_images](https://github.com/norlab-ulaval/libpointmatcher/assets/10325816/a40c9cc0-f4e8-4f26-95c8-608f3e2fcd17)

## Stacktrace (partial)
See build log in [attach](https://github.com/norlab-ulaval/libpointmatcher/files/13979654/Libpointmatcher_Bleeding_build_matrix_build_and_test_multi-arch_ci_PR_docker_images_14.1.log)

```stacktrace
10:46:12         #12 87.26 In file included from /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/utils/sparsetv.h:183,
10:46:12         #12 87.26                  from /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/SpectralDecomposition.h:38,
10:46:12         #12 87.26                  from /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/SpectralDecomposition.cpp:35:
10:46:12         #12 87.26 /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/utils/sparsetv.hpp:176: warning: ignoring ‘#pragma omp parallel’ [-Wunknown-pragmas]
10:46:12         #12 87.26   176 | #pragma omp parallel for
10:46:12         #12 87.26       |
10:46:12         #12 87.26 /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/utils/sparsetv.hpp:390: warning: ignoring ‘#pragma omp parallel’ [-Wunknown-pragmas]
10:46:12         #12 87.26   390 | #pragma omp parallel for
10:46:12         #12 87.26       |
10:46:12         #12 87.26 /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/utils/sparsetv.hpp:435: warning: ignoring ‘#pragma omp parallel’ [-Wunknown-pragmas]
10:46:12         #12 87.26   435 | #pragma omp parallel for
10:46:12         #12 87.26       |
10:46:12         #12 87.26 /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/utils/sparsetv.hpp:489: warning: ignoring ‘#pragma omp parallel’ [-Wunknown-pragmas]
10:46:12         #12 87.26   489 | #pragma omp parallel for
10:46:12         #12 87.26       |
10:46:12         #12 87.29 /opt/percep3d_libraries/libpointmatcher-build-system/pointmatcher/DataPointsFilters/SpectralDecomposition.cpp:145: warning: ignoring ‘#pragma omp parallel’ [-Wunknown-pragmas]
10:46:12         #12 87.29   145 |         #pragma omp parallel for
10:46:12         #12 87.29       |
10:47:01         #12 136.4 [ 76%] Linking CXX static library libpointmatcher.a
10:47:08         #12 143.9 [ 76%] Built target pointmatcher
10:47:08         #12 143.9 [ 76%] Building CXX object examples/CMakeFiles/align_sequence.dir/align_sequence.cpp.o
10:47:08         #12 143.9 [ 77%] Building CXX object examples/CMakeFiles/list_modules.dir/list_modules.cpp.o
10:47:08         #12 143.9 [ 78%] Building CXX object examples/CMakeFiles/build_map.dir/build_map.cpp.o
10:47:08         #12 143.9 [ 79%] Building CXX object examples/CMakeFiles/icp_customized.dir/icp_customized.cpp.o
10:47:08         #12 143.9 [ 80%] Building CXX object examples/CMakeFiles/pmicp.dir/icp.cpp.o
10:47:08         #12 143.9 [ 81%] Building CXX object examples/CMakeFiles/icp_simple.dir/icp_simple.cpp.o
10:47:08         #12 143.9 [ 82%] Building CXX object utest/CMakeFiles/utest.dir/ui/Matcher.cpp.o
10:47:08         #12 143.9 [ 82%] Building CXX object utest/CMakeFiles/utest.dir/ui/DataFilters.cpp.o
10:47:08         #12 143.9 [ 83%] Building CXX object utest/CMakeFiles/utest.dir/ui/IO.cpp.o
10:47:08         #12 144.0 [ 84%] Building CXX object utest/CMakeFiles/utest.dir/utest.cpp.o
10:47:08         #12 144.0 [ 85%] Building CXX object examples/CMakeFiles/icp_advance_api.dir/icp_advance_api.cpp.o
10:47:08         #12 144.0 [ 86%] Building CXX object examples/CMakeFiles/compute_overlap.dir/compute_overlap.cpp.o
10:47:14         #12 149.9 [ 87%] Linking CXX executable icp_simple
10:47:15         #12 150.2 [ 88%] Building CXX object utest/CMakeFiles/utest.dir/ui/Outliers.cpp.o
10:47:16         #12 151.5 [ 89%] Linking CXX executable list_modules
10:47:19         #12 154.4 [ 90%] Linking CXX executable align_sequence
10:47:19         #12 154.6 [ 90%] Linking CXX executable icp_customized
10:47:20         #12 155.8 [ 91%] Linking CXX executable build_map
10:47:20         #12 156.0 [ 92%] Building CXX object utest/CMakeFiles/utest.dir/ui/ErrorMinimizers.cpp.o
10:47:21         #12 156.3 [ 93%] Linking CXX executable compute_overlap
10:47:21         #12 157.0 [ 94%] Building CXX object utest/CMakeFiles/utest.dir/ui/Transformations.cpp.o
10:47:24         #12 160.0 [ 94%] Linking CXX executable pmicp
10:47:26         #12 161.2 [ 95%] Linking CXX executable icp_advance_api
10:47:27         #12 162.8 [ 96%] Building CXX object utest/CMakeFiles/utest.dir/ui/DataPoints.cpp.o
10:47:27         #12 162.9 [ 96%] Built target icp_simple
10:47:27         #12 162.9 [ 96%] Building CXX object utest/CMakeFiles/utest.dir/ui/Inspectors.cpp.o
10:47:29         #12 164.5 [ 96%] Built target list_modules
10:47:29         #12 164.5 [ 97%] Building CXX object utest/CMakeFiles/utest.dir/ui/Loggers.cpp.o
10:47:29         #12 164.6 [ 98%] Building CXX object utest/CMakeFiles/utest.dir/ui/icp/GeneralTests.cpp.o
10:47:30         #12 166.0 In file included from /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../utest.h:6,
10:47:30         #12 166.0                  from /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/Transformations.cpp:1:
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h: In instantiation of ‘void testing::internal::DefaultPrintTo(testing::internal::IsContainer, testing::internal::false_type, const C&, std::ostream*) [with C = Eigen::Matrix<long int, -1, -1>; testing::internal::IsContainer = int; testing::internal::false_type = testing::internal::bool_constant<false>; std::ostream = std::basic_ostream<char>]’:
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:9567:17:   required from ‘void testing::internal::PrintTo(const T&, std::ostream*) [with T = Eigen::Matrix<long int, -1, -1>; std::ostream = std::basic_ostream<char>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:9791:12:   required from ‘static void testing::internal::UniversalPrinter<T>::Print(const T&, std::ostream*) [with T = Eigen::Matrix<long int, -1, -1>; std::ostream = std::basic_ostream<char>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:9947:30:   required from ‘void testing::internal::UniversalPrint(const T&, std::ostream*) [with T = Eigen::Matrix<long int, -1, -1>; std::ostream = std::basic_ostream<char>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:9875:19:   required from ‘static void testing::internal::UniversalTersePrinter<T>::Print(const T&, std::ostream*) [with T = Eigen::Matrix<long int, -1, -1>; std::ostream = std::basic_ostream<char>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:10040:44:   required from ‘std::string testing::PrintToString(const T&) [with T = Eigen::Matrix<long int, -1, -1>; std::string = std::__cxx11::basic_string<char>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:18771:36:   required from ‘static std::string testing::internal::FormatForComparison<ToPrint, OtherOperand>::Format(const ToPrint&) [with ToPrint = Eigen::Matrix<long int, -1, -1>; OtherOperand = Eigen::Matrix<long int, -1, -1>; std::string = std::__cxx11::basic_string<char>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:18846:45:   required from ‘std::string testing::internal::FormatForComparisonFailureMessage(const T1&, const T2&) [with T1 = Eigen::Matrix<long int, -1, -1>; T2 = Eigen::Matrix<long int, -1, -1>; std::string = std::__cxx11::basic_string<char>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:18871:53:   required from ‘testing::AssertionResult testing::internal::CmpHelperEQ(const char*, const char*, const T1&, const T2&) [with T1 = Eigen::Matrix<long int, -1, -1>; T2 = Eigen::Matrix<long int, -1, -1>]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:18897:23:   required from ‘static testing::AssertionResult testing::internal::EqHelper<lhs_is_null_literal>::Compare(const char*, const char*, const T1&, const T2&) [with T1 = Eigen::Matrix<long int, -1, -1>; T2 = Eigen::Matrix<long int, -1, -1>; bool lhs_is_null_literal = false]’
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/Transformations.cpp:62:5:   required from here
10:47:30         #12 166.0 /opt/percep3d_libraries/libpointmatcher-build-system/utest/ui/../../contrib/gtest/gtest.h:9466:35: error: variable or field ‘it’ declared void
10:47:30         #12 166.0  9466 |   for (typename C::const_iterator it = container.begin();
10:47:30         #12 166.0       |                                   ^~
10:47:32         #12 167.5 make[2]: *** [utest/CMakeFiles/utest.dir/build.make:160: utest/CMakeFiles/utest.dir/ui/Transformations.cpp.o] Error 1
10:47:32         #12 167.5 make[2]: *** Waiting for unfinished jobs....
10:47:32         #12 167.8 [ 98%] Built target align_sequence
10:47:32         #12 167.9 [ 98%] Built target icp_customized
10:47:33         #12 168.5 [ 98%] Built target build_map
10:47:33         #12 168.8 [ 98%] Built target compute_overlap
10:47:35         #12 170.9 [ 98%] Built target pmicp
10:47:36         #12 171.6 [ 98%] Built target icp_advance_api
10:47:44         #12 179.8 make[1]: *** [CMakeFiles/Makefile2:553: utest/CMakeFiles/utest.dir/all] Error 2
10:47:44         #12 179.8 make: *** [Makefile:146: all] Error 2
10:47:44         #12 ERROR: process "/bin/bash -c source lpm_install_libpointmatcher_ubuntu.bash     --libpointmatcher-version ${LIBPOINTMATCHER_VERSION}     ${LIBPOINTMATCHER_INSTALL_SCRIPT_FLAG}" did not complete successfully: exit code: 2
10:47:44         ------
10:47:44          > [ci_build_and_test-amd64  7/10] RUN source lpm_install_libpointmatcher_ubuntu.bash     --libpointmatcher-version head     --build-system-CI-install --compile-test:
10:47:44         167.5 make[2]: *** [utest/CMakeFiles/utest.dir/build.make:160: utest/CMakeFiles/utest.dir/ui/Transformations.cpp.o] Error 1
10:47:44         167.5 make[2]: *** Waiting for unfinished jobs....
10:47:44         167.8 [ 98%] Built target align_sequence
10:47:44         167.9 [ 98%] Built target icp_customized
10:47:44         168.5 [ 98%] Built target build_map
10:47:44         168.8 [ 98%] Built target compute_overlap
10:47:44         170.9 [ 98%] Built target pmicp
10:47:44         171.6 [ 98%] Built target icp_advance_api
10:47:44         179.8 make[1]: *** [CMakeFiles/Makefile2:553: utest/CMakeFiles/utest.dir/all] Error 2
10:47:44         179.8 make: *** [Makefile:146: all] Error 2
10:47:44         ------
10:47:44         failed to solve: process "/bin/bash -c source lpm_install_libpointmatcher_ubuntu.bash     --libpointmatcher-version ${LIBPOINTMATCHER_VERSION}     ${LIBPOINTMATCHER_INSTALL_SCRIPT_FLAG}" did not complete successfully: exit code: 2
10:47:44         Process exited with code 17
10:47:45   Process exited with code 17 (Step: Build and test matrix › libpointmatcher ci_build_and_test-amd64 (Command Line))
10:47:45   Step Build and test matrix › libpointmatcher ci_build_and_test-amd64 (Command Line) failed
```

**Comment by RedLeader962** [comment]
Fixed as of [Build #17](http://132.203.26.125:8111/buildConfiguration/LibpointmatcherBuildSystem_BuildMatrixStatusBuildAndTestMultiArchCiBuildAndTestDockerImages/4002) run post [release  1.4.0](https://github.com/norlab-ulaval/libpointmatcher/releases/tag/1.4.0)

ToDo before closing the issue
- [ ] Add ubuntu jammy to CI build matrix
- [ ] mod doc current OS version support

**Comment by RedLeader962** [comment]
Fixed 
Special thanks to @boxanm and @simonpierredeschenes for solving the issue
